frontend/src/components/common/PrintCSVButton.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useEffect, useState } from "react";
+import {
+  Icon,
+  IconButton,
+  Box,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalBody,
+  FormControl,
+  Button,
+  FormHelperText,
+  Flex,
+  ScaleFade,
+  Alert,
+  AlertDescription,
+  AlertIcon,
+} from "@chakra-ui/react";
+import { AiFillPrinter } from "react-icons/ai";
+import { RangeDatepicker } from "chakra-dayzed-datepicker";
+import commonAPIClient from "../../APIClients/CommonAPIClient";
+import CSVConverter from "../../helper/CSVConverter";
+
+const PrintCSVButton = (): React.ReactElement => {
+  const [selectedDates, setSelectedDates] = useState<Date[]>([]);
+  const [showAlert, setShowAlert] = useState(false);
+
+  const handleClear = () => {
+    setSelectedDates([]);
+  };
+
+  const [isOpen, setOpen] = useState(false);
```

Comment:
nit: Can we move all uses of useState to the top of the function, and all handlers below? (i.e. in this case just swap the ordering of these 2)